### PR TITLE
Support non-bean naming convention

### LINKED ISSUE #25: Feature request: Drop requirement for 'get' prefix on methods
While it's easy to follow the convention in new code, it's a blocker to using @FreeBuilder in established classes. It's also a contentious requirement keeping many from adoption.

### PR DESCRIPTION
**Pull request by chrisalice**
Allow the user to drop the 'get'/'is' prefixes from their value type declaration. When using this "prefixless" convention, we drop the "get" and "set" prefixes from the generated Builder API to match. The existing behaviour is now referred to as "bean convention".

Code to detect and understand the naming convention is in the `o.i.f.processor.naming` package. Most code generation changes are localised to the BuilderMethods utility class we factored out for this purpose. To keep Jackson usage simple, we now automatically annotate getter methods on the value type; previously, Jackson was happily spotting the 'get'/'is' prefixes itself.

New tests are all copies of existing tests modified to use prefixless convention.

This closes #25.

### CODE REVIEW

**Comment by chrisalice** [comment]
Docs changes coming.

File: src/main/java/org/inferred/freebuilder/processor/naming/BeanConvention.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+  private final Messager messager;
+  private final Types types;
+
+  /**
+   * Verifies {@code method} is an abstract getter following the JavaBean convention. Any
+   * deviations will be logged as an error.
+   */
+  @Override
+  public Property.Builder getPropertyNamesOrNull(TypeElement valueType, ExecutableElement method) {
```
Comment (j-baker) [inline src/main/java/org/inferred/freebuilder/processor/naming/BeanConvention.java]: Optional

File: src/main/java/org/inferred/freebuilder/processor/naming/NamingConventions.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        beanMethod = firstNonNull(beanMethod, method);
+        break;
+      case PREFIXLESS:
+        prefixlessMethod = firstNonNull(prefixlessMethod, method);
+        break;
+      default:
+        break;
+      }
+    }
+    if (prefixlessMethod != null) {
```
Comment (j-baker) [inline src/main/java/org/inferred/freebuilder/processor/naming/NamingConventions.java]: && beanMethod != null?

Comment (chrisalice) [inline src/main/java/org/inferred/freebuilder/processor/naming/NamingConventions.java]: That's on the next line.

File: src/main/java/org/inferred/freebuilder/processor/naming/BeanConvention.java (hunk 1 above)
Comment (chrisalice) [inline src/main/java/org/inferred/freebuilder/processor/naming/BeanConvention.java]: Done, thanks!

File: src/main/java/org/inferred/freebuilder/processor/naming/NamingConventions.java (hunk 2 above)
Comment (j-baker) [inline src/main/java/org/inferred/freebuilder/processor/naming/NamingConventions.java]: yes - it just felt like doing

    if (cond1) {
        if (cond2) {

        }
    }

could be more terse.

Comment (j-baker) [inline src/main/java/org/inferred/freebuilder/processor/naming/NamingConventions.java]: misread.

**Pull request merged**